### PR TITLE
Add upgrade flags to pip installs

### PR DESCRIPTION
**Pull request by dakinggg**
# What does this PR do?
Adds `--upgrade` to the `pip install`s in CI. This ensures that CI will flag a dependency that upgrade that breaks composer

Closes [CO-1707](https://mosaicml.atlassian.net/browse/CO-1707)

[CO-1707]: https://mosaicml.atlassian.net/browse/CO-1707?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/pytest-cpu.yaml
Comment (mvpatel2000) [inline .github/workflows/pytest-cpu.yaml:51]:
```suggestion
          python -m pip install --upgrade .[all]
```

File: .github/workflows/code-quality.yaml
Comment (mvpatel2000) [inline .github/workflows/code-quality.yaml:43]:
```suggestion
          python -m pip install --upgrade .${{ matrix.pip_deps }}
```

**Review by mvpatel2000 — COMMENTED** [review]
There's a few more hanging `-e` that I added when copy pasting things which shouldn't really be there (since we install after checking out the right branch). Can you remove them in this PR? 

Will approve after that change + merging with changes to `mcp_pytest`

File: .github/workflows/pytest-daily.yaml
Comment (mvpatel2000) [inline .github/workflows/pytest-daily.yaml:74]:
```suggestion
          python -m pip install --upgrade .[all]
```

**Review by mvpatel2000 — APPROVED** [review]
LGTM. Can you remove the `-e` I left in

File: .github/workflows/pytest-daily.yaml
Comment (dakinggg) [reply .github/workflows/pytest-daily.yaml:74]: oops, i meant to do that. my fingers just typed something else.

**Pull request merged**
